src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddMemberToInterface.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    public class CannotAddMemberToInterface : Rule
+    {
+        public override void Initialize(RuleRunnerContext context)
+        {
+            // StrictMode scenario should be handled by MembersMustExist rule.
+            if (!Settings.StrictMode)
+            {
+                context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+            }
+        }
+
+        private void RunOnMemberSymbol(ISymbol left, ISymbol right, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            if (left == null && right != null && right.ContainingType.TypeKind == TypeKind.Interface)
```

Comment:
why is ```left == null``` here ?

Reply:
That means the member is added to right. So it doesn't exist on left. That is what this rule is trying to catch, when a member is added to right (does not exist on left) and the type is an interface.